logic/src/commonMain/kotlin/com/wire/kalium/logic/sync/receiver/conversation/message/ApplicationMessageHandler.kt

```diff
@@ -196,47 +198,46 @@ internal class ApplicationMessageHandlerImpl(
         }
     }
 
-    private suspend fun processMessage(message: Message.Regular) {
+    private suspend fun processMessage(message: Message.Regular, expireAfterMillis: Long? = null) {
         logger.i(message = "Message received: { \"message\" : $message }")
 
         when (val content = message.content) {
             // Persist Messages - > lists
-            is MessageContent.Text -> handleTextMessage(message, content)
-
-            is MessageContent.FailedDecryption -> {
-                persistMessage(message)
-            }
-
-            is MessageContent.Knock -> handleKnock(message)
+            is MessageContent.Text -> persistRegularMessage(adjustTextMessageWithQoutedReference(message, content), expireAfterMillis)
+            is MessageContent.FailedDecryption -> persistRegularMessage(message, expireAfterMillis)
+            is MessageContent.Knock -> persistRegularMessage(message, expireAfterMillis)
             is MessageContent.Asset -> assetMessageHandler.handle(message, content)
-
+            is MessageContent.RestrictedAsset -> TODO()
             is MessageContent.Unknown -> {
                 logger.i(message = "Unknown Message received: $message")
-                persistMessage(message)
+                persistRegularMessage(message, expireAfterMillis)
             }
 
-            is MessageContent.RestrictedAsset -> TODO()
         }
     }
 
-    private suspend fun handleKnock(message: Message.Regular) {
-        persistMessage(message)
+    private suspend fun persistRegularMessage(message: Message.Regular, expireAfterMillis: Long? = null) {
+        if (expireAfterMillis != null) {
+            persistMessage(Message.Ephemeral(expireAfterMillis, null, message))
```

Comment:
I thought `Message.Ephemeral` was removed flattened. Is something not pushed?

Reply:
oops I forgot about that part